fuel/datasets/binarized_mnist.py

```diff
@@ -1,19 +1,14 @@
 # -*- coding: utf-8 -*-
 import logging
 import os
-from collections import OrderedDict
-
-import numpy
 
 from fuel import config
-from fuel.datasets import IndexableDataset
-from fuel.utils import do_not_pickle_attributes
+from fuel.datasets import H5PYDataset
 
 logger = logging.getLogger(__name__)
 
 
-@do_not_pickle_attributes('indexables')
-class BinarizedMNIST(IndexableDataset):
+class BinarizedMNIST(H5PYDataset):
```

Comment:
Woops. My first paragraph went unfinished. What I'm mainly against is h5py-specific implementation details leaking into dataset classes. That isn't happening here, so no big deal as far as I'm concerned.